src/hotspot/share/gc/g1/g1RootClosures.cpp

```diff
@@ -50,10 +50,10 @@ class G1EvacuationClosures : public G1EvacuationRootClosures {
 // Closures used during concurrent start.
 // The treatment of "weak" roots is selectable through the template parameter,
 // this is usually used to control unloading of classes and interned strings.
-template <G1Mark MarkWeak>
+template <bool should_mark_weak>
 class G1ConcurrentStartMarkClosures : public G1EvacuationRootClosures {
-  G1SharedClosures<G1MarkFromRoot> _strong;
-  G1SharedClosures<MarkWeak>       _weak;
+  G1SharedClosures<true>        _strong;
+  G1SharedClosures<should_mark_weak>   _weak;
```

Comment:
Sorry for being picky, but now the alignment of the variables are off. I would prefer:
```suggestion
  G1SharedClosures<true>             _strong;
  G1SharedClosures<should_mark_weak> _weak;
```